x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -653,8 +656,22 @@ protected LogicalPlan rule(Enrich re) {
         }
     }
 
-    protected static class PushDownAndCombineOrderBy extends OptimizerRules.OptimizerRule<OrderBy> {
+    protected static class PushDownMvExpand extends OptimizerRules.OptimizerRule<MvExpand> {
```

Comment:
What's the advantage of pushing down MvExpand? I'd argue we want the opposite as this increases the amount of data earlier in the pipeline.